crates/iroha_test_network/src/lib.rs

```diff
@@ -280,12 +282,29 @@ impl Network {
     }
 }
 
+/// Determines how [`NetworkBuilder`] configures [`SmartContractParameter::Fuel`] in the genesis.
+#[derive(Default)]
+pub enum WasmFuelConfig {
+    /// Do not set anything, i.e. let Iroha use its default value
+    #[default]
+    Unset,
+    /// Set to a specific value
+    Value(NonZero<u64>),
+    /// Determine automatically based on the WASM samples build profile
+    /// (received from [`iroha_test_samples::load_wasm_build_profile`]).
+    ///
+    /// If the profile is not optimized, the fuel will be increased, otherwise the same as
+    /// [`WasmFuelConfig::Unset`].
+    Auto,
+}
```

Comment:
@aoyako could you please take a look?

I heard you had some ideas about how to handle dynamic fuel limits, and you previously expressed some thoughts on how to set fuel limit in tests.

Reply:
IMO, now it looks much better than it was before. I think `Auto` should be the default to avoid repeating and not accidentally forgetting `with_wasm_fuel`.

Reply:
Oh, well, okay!

I didn't make `Auto` the default because it will increase wasm fuel for tests that are currently passing without this increase.

Reply:
Set `Auto` as a default.